client/my-sites/email/mailboxes/index.tsx

```diff
@@ -139,6 +139,7 @@ const MailboxesManagement = ( {
 					selectedIntervalLength={ selectedIntervalLength }
 					showActiveDomainList={ showActiveDomainList( nonWPCOMDomains ) }
 					source={ MAILBOXES_SOURCE }
+					enableSetupRedirects={ false }
```

Comment:
I'm not convinced this is a very good solution. 

I think a better approach would be to look for a `new-email` URL query parameter in this component. If we find one, we should force-refresh data returned from `getDomainsBySiteId`. That way, `hasAtLeastOneMailbox` would return a correct value, and `EmailHome` wouldn't redirect us away from `/mailboxes`.

Does that make sense?